src/genie/libs/parser/iosxr/show_dampening.py

```diff
@@ -0,0 +1,257 @@
+
+# parser utils
+from genie.libs.parser.utils.common import Common
+
+# =============================================
+# Parser for 'show im dampening'
+# =============================================
+
+class ShowImDampeningSchema(MetaParser):
+    schema = {
+                'interface': {
+                    Any(): {
+                    'index': {
+                        Any(): {
+                        Optional('capsulation'): str,
+                        'penalty': int,
+                        'suppressed': str,
+                        Optional('protocol'): str,
+                        }
+                    }
+                }
+            }
+        }
+                    
+class ShowImDampening(ShowImDampeningSchema):
+    """ Parser for show im dampening"""
+    
+    cli_command = 'show im dampening'
+
+    def cli(self,output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+        
+           
+        result_dict = {}
+        index=0
+        interface=""       
+        # GigabitEthernet0/0/0/1                                               2389 YES
+        # POS0/2/0/0                  <base>             ppp                      0 NO 
+        p1 = re.compile(r'^(?P<interface>\S+) +(?P<prot>\S+)? +(?P<cap>\S+)? +(?P<pen>\d+) +(?P<sup>YES|NO)')
+
+        for line in out.splitlines():
+            if line:
+                line = line.strip()
+            else:
+                continue
+
+            # GigabitEthernet0/0/0/1                                               2389 YES
+            # POS0/2/0/0                  <base>             ppp                      0 NO             
+
+            m = p1.match(line)
+            
+            if m:
+                group = m.groupdict()
+                
+                if interface==group['interface']:
+                    interface=group['interface']
+                    index=index+1
+                else:
+                    interface=group['interface']
+                    index=1   
+                protocol_dict = result_dict.setdefault('interface', {}).setdefault(interface, {}).setdefault('index', {}).setdefault(index, {})
+                protocol_dict.update({'penalty': int(group['pen'])})
+                protocol_dict.update({'suppressed': group['sup']})
+                if group['cap']:
+                    protocol_dict.update({'capsulation': group['cap']})
+                if group['prot']:
+                    protocol_dict.update({'protocol': group['prot']})
+                continue
+   
+        return result_dict
+
+class ShowImDampeningIntfSchema(MetaParser):
+    schema = {
+        'interface': {
+            Any(): {
+            'dampening_status': str,
```

Comment:
Please fix the indentation for this schema

Reply:
Fixed